api/versions.go

```diff
@@ -1174,7 +1175,73 @@ func (api *DatasetAPI) putState(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if stateUpdate.State == models.PublishedState && currentVersion.Distributions != nil && len(*currentVersion.Distributions) > 0 {
+		err = api.publishDistributionFiles(ctx, currentVersion, logData)
+		if err != nil {
+			log.Error(ctx, "putState endpoint: failed to publish distribution files", err, logData)
```

Comment:
Do we still want to return a 200 response if one or more distributions fails to publish?

Reply:
Good point, I think we should return an error if any of the files fail to publish. I've updated it 👍